frontend/lit/TagReferences/Main.js

```diff
@@ -142,32 +143,54 @@ class TagReferencesMain extends Component {
                                                 ? "Resolved Tag: ".concat(tag.get_full_name())
                                                 : tag.get_full_name()
                                         }
-                                        className="refTag refTagEditing"
-                                        onClick={
+                                        className={
                                             store.config.conflict_resolution
-                                                ? null
-                                                : () => store.removeTag(tag)
-                                        }>
-                                        {this.state.showFullTag
-                                            ? tag.get_full_name()
-                                            : tag.data.name}
-                                    </span>
-                                ))}
-                                {selectedReferenceUserTags.map((tag, i) => (
-                                    <span
-                                        key={i}
-                                        title={tag.get_full_name()}
-                                        className="refTag refUserTag refTagEditing"
+                                                ? _.find(
+                                                      selectedReferenceUserTags,
+                                                      e => e.data.pk == tag.data.pk
+                                                  )
+                                                    ? "refTag refTagEditing"
+                                                    : "refTag refUserTagDiff refTagEditing"
+                                                : "refTag refTagEditing"
+                                        }
                                         onClick={
                                             store.config.conflict_resolution
-                                                ? () => store.removeTag(tag)
-                                                : null
+                                                ? _.find(
+                                                      selectedReferenceUserTags,
+                                                      e => e.data.pk == tag.data.pk
+                                                  )
+                                                    ? () => store.removeTag(tag)
+                                                    : () => store.addTag(tag)
+                                                : () => store.removeTag(tag)
                                         }>
                                         {this.state.showFullTag
                                             ? tag.get_full_name()
                                             : tag.data.name}
                                     </span>
                                 ))}
+                                {selectedReferenceUserTags
```

Comment:
This has been simplified after other refactors; since its just a single line filter now I don't think it would make much sense to define it in the render method